File: nagasena/src/org/openexi/proc/io/compression/EXIEventElement.java
package org.openexi.proc.io.compression;

import org.openexi.proc.common.BinaryDataSource;
import org.openexi.proc.common.EventType;
import org.openexi.proc.common.EventDescription;
import org.openexi.schema.Characters;

final class EXIEventElement implements EventDescription {

  String prefix;
  EventType eventType;
  
  public EXIEventElement() {
    prefix = null;
    eventType = null;
  }
  
  ///////////////////////////////////////////////////////////////////////////
  // Implementation of EXIEvent interface
  ///////////////////////////////////////////////////////////////////////////

  public final byte getEventKind() {
    return EventDescription.EVENT_SE;
  }

  public String getName() {
    return eventType.name;
  }

  public String getURI() {
    return eventType.uri;
  }

  public int getNameId() {
    return eventType.getNameId();
  }

  public int getURIId() {
    return eventType.getURIId();
  }

  public String getPrefix() {
    return prefix;
  }
  
  public final Characters getCharacters() {
    return null;
  }
  
  public BinaryDataSource getBinaryDataSource() {
    return null;
  }

  public final EventType getEventType() {
    return eventType;
  }

}

File: nagasenacs/src/Nagasena/Proc/IO/Compression/EXIEventElement.cs
using BinaryDataSource = Nagasena.Proc.Common.BinaryDataSource;
using EventType = Nagasena.Proc.Common.EventType;
using EventDescription = Nagasena.Proc.Common.EventDescription;
using EventDescription_Fields = Nagasena.Proc.Common.EventDescription_Fields;
using Characters = Nagasena.Schema.Characters;

namespace Nagasena.Proc.IO.Compression {

  internal sealed class EXIEventElement : EventDescription {

    internal string prefix;
    internal EventType eventType;

    public EXIEventElement() {
      prefix = null;
      eventType = null;
    }

    ///////////////////////////////////////////////////////////////////////////
    // Implementation of EXIEvent interface
    ///////////////////////////////////////////////////////////////////////////

    public sbyte EventKind {
      get {
        return EventDescription_Fields.EVENT_SE;
      }
    }

    public string Name {
      get {
        return eventType.name;
      }
    }

    public string URI {
      get {
        return eventType.uri;
      }
    }

    public int NameId {
      get {
        return eventType.NameId;
      }
    }

    public int URIId {
      get {
        return eventType.URIId;
      }
    }

    public string Prefix {
      get {
        return prefix;
      }
    }

    public Characters Characters {
      get {
        return null;
      }
    }

    public BinaryDataSource BinaryDataSource {
      get {
        return null;
      }
    }

    public EventType getEventType() {
      return eventType;
    }

  }

}